File: src/test/java/recipestore/input/RecipeApiIntegrationTest.java
package recipestore.input;

import lombok.SneakyThrows;
import org.apache.jena.ext.com.google.common.io.Resources;
import org.apache.jena.rdf.model.Resource;
import org.hamcrest.Matchers;
import org.junit.BeforeClass;
import org.junit.Test;
import recipestore.db.triplestore.FileBasedTripleStoreDAO;

import java.io.InputStream;
import java.util.List;
import java.util.stream.Collectors;

import static org.hamcrest.MatcherAssert.assertThat;

public class RecipeApiIntegrationTest {

    private static RecipeApi recipeApi;

    @BeforeClass
    @SneakyThrows
    public static void setup() {
        final InputStream recipeStream = Resources.getResource("recipe.nq").openStream();
        recipeApi = new RecipeApi(new FileBasedTripleStoreDAO("int-test"), recipeStream);
        recipeApi.loadRecipe(true);
    }

    @Test
    public void shouldGetRecipeDataFromTheTripleStore() {

        final List<Resource> recipeData = recipeApi.getRecipeData(RecipeResourceFilter.getRecipeWithMinimumNumberOfRating)
                .collect(Collectors.toList());
        assertThat("Should have allrecipes records ", recipeData.size(), Matchers.greaterThan(1));


    }


}

File: src/main/scala/recipestore/input/RecipeApiIntegrationTest.scala
package recipestore.input

import java.io.InputStream

import org.apache.jena.ext.com.google.common.io.Resources
import org.apache.jena.rdf.model.Resource
import org.scalatest.{BeforeAndAfter, FunSuite, Matchers}
import recipestore.db.triplestore.FileBasedTripleStoreDAO

object RecipeApiIntegrationTest extends FunSuite with Matchers with BeforeAndAfter {
  private var recipeApi: RecipeApi = null

  def before() {
    val recipeStream: InputStream = Resources.getResource("recipe.nq").openStream
    recipeApi = new RecipeApi(new FileBasedTripleStoreDAO("int-test"), recipeStream)
    recipeApi.loadRecipe(true)
  }

  def shouldGetRecipeDataFromTheTripleStore() {
    val recipeData: Seq[Resource] = recipeApi.getRecipeData(RecipeResourceFilter.getRecipeWithMinimumNumberOfRating)
    recipeData.size should be > 1
  }
}
